### ISSUE TITLE
Add support for tenant-level Deployment Stacks

### ISSUE THREAD

**Issue by aperezmicrosoft** [Azure/deployment-stacks#144]
**Is your feature request related to a problem? Please describe.**
Enable ability to Create/Deploy Deployment Stacks at Tenant scope.

**Describe the solution you'd like**
Tenant-level deployment stacks support.

**Additional context**
N/A

**Comment by snarkywolverine** [comment]
For those who would like to see this happen - please elaborate on the scenario.  We are curious what makes the tenant scope more valuable than deploying directly to a management group (top level or otherwise).

**Comment by marsontret** [comment]
Isn't this related to #156?

**Comment by snarkywolverine** [comment]
@marsontret No. #156 addresses how management groups are handled when deployed by a stack (e.g, at another management group scope).

Customers who wish to _create the stack_ at tenant scope should upvote this.

Does that make sense?

**Comment by veikkoeeva** [comment]
> For those who would like to see this happen - please elaborate on the scenario. We are curious what makes the tenant scope more valuable than deploying directly to a management group (top level or otherwise).

I was thinking to bootsrap an Azure tenant. First create a root management group in tenant, assing a policy to it that restrict federated identities to a certain repository. When a federated identity is created and assigned to a CI pipeline, it then deployes the rest of the Azure tenant structure using Bicep and bicepparm files.

It would be great to be able to manage this bootstrap phase before CI system takes over as a singular unit also.

**Comment by gXkch** [comment]
My use-case is that we deploy many policy assignments across multiple management groups with one big tenant level deployment. I would like to have the possibility to have tenant level stacks so when I delete a policy assignment in my bicep file / parameter file, it will also be deleted from my Azure environment.

**Comment by PMDev483** [comment]
My use case is to let our customers manage the integration between my company's SaaS product and their Azure environment. We need to integrate at the tenant, MG, or Subscription level depending on the customer. The level of our integration can also change over time (e.g. a customer starts out giving us access to one Subscription and over time expands to include the entire Azure environment). For tenancy-wide deployments, there are items like policies that do not fit into an MG deployment. This makes managing the integration and removing everything (such as after a POV) more involved than it should be.
